### DISCUSSION TITLE
Deploying experimental App Directory

### DISCUSSION THREAD

**Discussion by etherealheim** [vercel/community#2293] [Help] [answered]
### Summary

Hey, I’m having issues with deploying newest app directory, it gives me 404s even when I try to put some settings in "output directory" like ".next", ".", "/app" but nothing seems to be working, was looking on internet and also asked gpt, but nothing.

### Example

https://form-project-1zdfkw79y-etherealheim.vercel.app/

**Comment by TheoKondak** [comment]
Hi!

I suggest adding some extra information about the project, and if possible, a minimal reproduction of the issue.

What do you mean by: 

> when I try to put some settings in "output directory" like ".next", ".", "/app" 

You should not change `.next` manually. It's a dynamically generated folder.

Also, have you checked the beta docs about how the new `app` folder routing works? In case you have missed it here is a link explaining the correct file/folder structure and file naming.

[See the docs](https://beta.nextjs.org/docs/routing/fundamentals)
And a youtube video from Vercel [Next.js App Router: Routing, Data Fetching, Caching](https://youtu.be/gSSsZReIFRk)

**Comment by etherealheim** [comment]
Hi Theo,

hm, I went through the beta docs, thanks for them, but there seems to be disabled section for **deploying** "new" app directory. 

Thats basically my issue. Let’s ignore the:

> You should not change .next manually. It's a dynamically generated folder.

I think I misunderstood what it actually does. So my mistake.

So I will talk about clean and simple app directory with one index page and one subpage. When I try to deploy this "beta" I get this 404 error. So the issue is: the deployment goes through but when I want to visit my app it shows

<img width="1106" alt="Screenshot 2023-05-04 at 14 26 05" src="https://user-images.githubusercontent.com/128578761/236203624-27050cb8-0e28-439d-9c2a-8d17ceed3442.png">

Cannot provide anything more as far as I can think of.

TL;DR - Experimental app directory shows 404 with default settings when deploying on Vercel.

**Reply by TheoKondak** [reply]
Hey,

So how does your folder structure look like?

Does it look like the following?

```
root/
  app/
    layout.tsx
    page.tsx
    not-found.tsx
    about/
      page.tsx
    blog/
      [slug]/
        page.tsx
```

**Comment by etherealheim** [comment]
```
root/
  app/
      api/
      components/
            /...
            Navigation.tsx
            NavigationLinks.tsx
      project/
            page.tsx
      layout.tsx
      page.tsx
      public/
```

I don’t have 404 defined yet, but I don’t think its relevant. First I need to render index page as it normally worked in stable old directory

next.config.js
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    appDir: true,
  },
}

module.exports = nextConfig
```

**Reply by TheoKondak** [reply]
I think you should not have the `components/` and `public/` folders like this. Try to move them outside `app/`.

Also, the `api/` could also be a problem depending on how it is implemented. Have a look [here](https://beta.nextjs.org/docs/data-fetching/api-routes). Personally, I still use the `pages/` for my API endpoints.

Have you tried keeping only the basic files, like `layout.tsx` and `pages.tsx` in your project without the other routes? Does it work like that?

**Reply by etherealheim** [reply, marked as answer]
I created new project and moved all pages in similar fashion as I started and with release of 13.4 it seemed to solve it. I still don’t have idea what was wrong, I tried the docs tips, yours as well, I moved stuff, I removed "use client"; created pages folder. Nothing. Now its deployed and working. Luckily it is my small project still exploring stuff so it didn’t hurt :) Thanks for your help Theo.
